app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/facet/standard/AppEngineStandardSupportProvider.java

```diff
@@ -128,8 +144,15 @@ private void addSupport(
       Set<AppEngineStandardMavenLibrary> librariesToAdd) {
     FacetType<AppEngineStandardFacet, AppEngineStandardFacetConfiguration> facetType =
         AppEngineStandardFacet.getFacetType();
+
     AppEngineStandardFacet appEngineStandardFacet =
-        FacetManager.getInstance(module).addFacet(facetType, facetType.getDefaultFacetName(), null);
+        AppEngineStandardFacet.getAppEngineFacetByModule(module);
+    if (appEngineStandardFacet == null) {
```

Comment:
We added this check since we now allow adding support as many times as user wants, correct? Previously it was guarded by menu item check?

Reply:
exactly